src/compiler/scanner/lines_0_1000.rs

```diff
@@ -96,73 +940,644 @@ pub(super) fn is_code_point(code: u32) -> bool {
+                }
+            }
+            _ => {
+                if ch > CharacterCodes::max_ascii_character && is_white_space_like(ch) {
+                    pos += 1;
+                    continue;
+                }
+            }
         }
         return isize::try_from(pos).unwrap();
     }
 }
 
-pub(super) fn is_identifier_start(ch: char) -> bool {
+// lazy_static! {
+//     pub(super) static ref merge_conflict_marker_length: usize = "<<<<<<<".len();
+// }
+pub(super) const fn merge_conflict_marker_length() -> usize {
+    // "<<<<<<<".len()
+    7
+}
+
+fn is_conflict_marker_trivia(text: &SourceTextAsChars, pos: usize) -> bool {
+    // Debug_.assert(pos >= 0);
+
+    if pos == 0 || matches!(maybe_text_char_at_index(text, pos - 1), Some(ch) if is_line_break(ch))
+    {
+        let ch = maybe_text_char_at_index(text, pos);
+
+        if pos + merge_conflict_marker_length() < text_len(text) {
+            let ch = ch.unwrap();
+            for i in 0..merge_conflict_marker_length() {
+                if text_char_at_index(text, pos + i) != ch {
+                    return false;
+                }
+            }
+
+            return ch == CharacterCodes::equals
+                || text_char_at_index(text, pos + merge_conflict_marker_length())
+                    == CharacterCodes::space;
+        }
+    }
+
+    false
+}
+
+fn scan_conflict_marker_trivia<TError: FnMut(&DiagnosticMessage, Option<usize>, Option<usize>)>(
+    text: &SourceTextAsChars,
+    mut pos: usize,
+    error: Option<TError>,
+) -> usize {
+    if let Some(mut error) = error {
+        error(
+            &Diagnostics::Merge_conflict_marker_encountered,
+            Some(pos),
+            Some(merge_conflict_marker_length()),
+        );
+    }
+
+    let ch = text_char_at_index(text, pos);
+    let len = text_len(text);
+
+    if matches!(ch, CharacterCodes::less_than | CharacterCodes::greater_than) {
+        while pos < len && !is_line_break(text_char_at_index(text, pos)) {
+            pos += 1;
+        }
+    } else {
+        Debug_.assert(
+            matches!(ch, CharacterCodes::bar | CharacterCodes::equals),
+            None,
+        );
+        while pos < len {
+            let current_char = text_char_at_index(text, pos);
+            if matches!(
+                current_char,
+                CharacterCodes::equals | CharacterCodes::greater_than
+            ) && current_char != ch
+                && is_conflict_marker_trivia(text, pos)
+            {
+                break;
+            }
+
+            pos += 1;
+        }
+    }
+
+    pos
+}
+
+lazy_static! {
+    pub(super) static ref shebang_trivia_regex: Regex = Regex::new(r"^#!.*").unwrap();
+}
+
+// pub(crate) fn is_shebang_trivia(
+//     text: &str,
+//     pos: usize, // this is a char index, not a string (byte) index, it's only ok because it's always 0 here
+// ) -> bool {
+//     Debug_.assert(pos == 0, None);
+//     shebang_trivia_regex.is_match(text)
```

Comment:
Mimicking the Typescript version by using a regex for this shebang-processing was necessitating passing around the `&str` version of the source text (since that's what you need to feed to a regex), and then eg for `scan_shebang_trivia()` to return a # of _chars_ processed instead of a # of _bytes_ processed I needed to explicitly convert that. So since it's a simple regex to mimic by hand I just rewrote these to take a `&SourceTextAsChars` instead